snuba/migrations/snuba_migrations/functions/0002_add_thread_id.py

```diff
@@ -0,0 +1,180 @@
+                    settings={"index_granularity": self.index_granularity},
+                    ttl="timestamp + toIntervalDay(retention_days)",
+                ),
+            ),
+            operations.CreateMaterializedView(
+                storage_set=self.storage_set,
+                view_name=self.local_view_table,
+                destination_table_name=self.local_materialized_table,
+                columns=columns,
+                query=self.__MATVIEW_STATEMENT,
+            ),
+        ]
+
+    def backwards_local(self) -> Sequence[operations.SqlOperation]:
+        # intentionally empty as this is a destructive migration and we do not want to go back
+        return []
+
+    def forwards_dist(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.FUNCTIONS,
+                table_name="functions_raw_dist",
+                column=Column("thread_id", UInt(64)),
+                after="profile_id",
+            ),
+            operations.DropTable(
+                storage_set=self.storage_set, table_name=self.dist_materialized_table
+            ),
+            operations.CreateTable(
+                storage_set=self.storage_set,
+                table_name=self.dist_materialized_table,
+                columns=columns,
+                engine=table_engines.Distributed(
+                    local_table_name=self.local_materialized_table,
+                    sharding_key=None,
+                ),
+            ),
+        ]
+
+    def backwards_dist(self) -> Sequence[operations.SqlOperation]:
+        # intentionally empty as this is a destructive migration and we do not want to go back
+        return []
+
+    @property
+    def __MATVIEW_STATEMENT(self) -> str:
+        return f"""
+            SELECT
+                project_id,
+                transaction_name,
+                name,
+                package,
+                path,
+                fingerprint,
+                parent_fingerprint,
+                depth,
+                is_application,
+                platform,
+                environment,
+                release,
+                os_name,
+                os_version,
+                toDateTime({self.data_granularity} * intDiv(toUnixTimestamp(timestamp), {self.data_granularity})) AS timestamp,
+                retention_days,
+                countState(arrayJoin(durations) AS duration) AS count,
+                quantilesState(0.5, 0.75, 0.9, 0.95, 0.99)(duration) AS percentiles,
+                minState(duration) AS min,
+                maxState(duration) AS max,
+                avgState(duration) AS avg,
+                sumState(duration) AS sum,
+                argMaxState((profile_id, thread_id), duration) as worst,
+                groupUniqArrayState(5)((profile_id, thread_id)) as examples
+            FROM {self.local_raw_table}
+            WHERE materialization_version = 1
```

Comment:
Bumping the materialization version here as we only want the rows that have `thread_id` populated